content/config/daemon/troubleshoot.md

```diff
@@ -88,15 +143,313 @@ happening, ensure that your application runs on hosts with adequate memory and
+
+You only need to use one of these methods.
+
+### Specify DNS servers for Docker
+
+The default location of the configuration file is `/etc/docker/daemon.json`. You
+can change the location of the configuration file using the `--config-file`
+daemon flag. The following instruction assumes that the location of the
+configuration file is `/etc/docker/daemon.json`.
+
+1. Create or edit the Docker daemon configuration file, which defaults to
+   `/etc/docker/daemon.json` file, which controls the Docker daemon
+   configuration.
+
+   ```console
+   $ sudo nano /etc/docker/daemon.json
+   ```
+
+2. Add a `dns` key with one or more DNS server IP addresses as values.
+
+   ```json
+   {
+     "dns": ["8.8.8.8", "8.8.4.4"]
+   }
+   ```
+
+   If the file has existing contents, you only need to add or edit the `dns`
+   line. If your internal DNS server can't resolve public IP addresses, include
+   at least one DNS server that can. Doing so allows you to connect to Docker
+   Hub, and your containers to resolve internet domain names.
+
+   Save and close the file.
+
+3. Restart the Docker daemon.
+
+   ```console
+   $ sudo service docker restart
+   ```
+
+4. Verify that Docker can resolve external IP addresses by trying to pull an
+   image:
+
+   ```console
+   $ docker pull hello-world
+   ```
+
+5. If necessary, verify that Docker containers can resolve an internal hostname
+   by pinging it.
+
+   ```console
+   $ docker run --rm -it alpine ping -c4 <my_internal_host>
+
+   PING google.com (192.168.1.2): 56 data bytes
+   64 bytes from 192.168.1.2: seq=0 ttl=41 time=7.597 ms
+   64 bytes from 192.168.1.2: seq=1 ttl=41 time=7.635 ms
+   64 bytes from 192.168.1.2: seq=2 ttl=41 time=7.660 ms
+   64 bytes from 192.168.1.2: seq=3 ttl=41 time=7.677 ms
+   ```
+
+### Turn off `dnsmasq`
+
+{{< tabs >}}
+{{< tab name="Ubuntu" >}}
+
+If you prefer not to change the Docker daemon's configuration to use a specific
+IP address, follow these instructions to turn off `dnsmasq` in NetworkManager.
+
+1. Edit the `/etc/NetworkManager/NetworkManager.conf` file.
+
+2. Comment out the `dns=dnsmasq` line by adding a `#` character to the beginning
+   of the line.
+
+   ```text
+   # dns=dnsmasq
+   ```
+
+   Save and close the file.
+
+3. Restart both NetworkManager and Docker. As an alternative, you can reboot
+   your system.
+
+   ```console
+   $ sudo systemctl restart network-manager
+   $ sudo systemctl restart docker
+   ```
+
+{{< /tab >}}
+{{< tab name="RHEL, CentOS, or Fedora" >}}
+
+To turn off `dnsmasq` on RHEL, CentOS, or Fedora:
+
+1. Turn off the `dnsmasq` service:
+
+   ```console
+   $ sudo systemctl stop dnsmasq
+   $ sudo systemctl disable dnsmasq
+   ```
+
+2. Configure the DNS servers manually using the
+   [Red Hat documentation](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/6/html/Deployment_Guide/s1-networkscripts-interfaces.html).
+
+{{< /tab >}}
+{{< /tabs >}}
+
+### Allow access to the remote API through a firewall
```

Comment:
Perhaps this is content that should be part of the "allow remote access" to the API 🤔

I'm slightly on the fence how much we should document this though, as exposing the API remotely is always a big security risk; more so because this section is thin on risks (e.g. it mentions "or 2375 if you're not using TLS - which effectively means "give the world root access" 😬) 🤔

Reply:
Yeah... I moved this section to the more appropriate doc (not really troubleshooting). That doc contains a strong warning about this already.